test/internal/k8sutils/utils_create.go

```diff
@@ -41,8 +53,36 @@ func MustCreateDaemonset(ctx context.Context, daemonsets typedappsv1.DaemonSetIn
 	return nil
 }
 
+func MustRestartDaemonset(ctx context.Context, daemonsets typedappsv1.DaemonSetInterface, ds appsv1.DaemonSet) error {
+	data := fmt.Sprintf(`{"spec": {"template": {"metadata": {"annotations": {"kubectl.kubernetes.io/restartedAt": "%q"}}}}}`, time.Now().Format("20060102150405"))
+	_, err := daemonsets.Patch(ctx, ds.Name, types.StrategicMergePatchType, []byte(data), metav1.PatchOptions{})
+	if err != nil {
+		return errors.Wrap(err, "failed to patch daemonset")
+	}
+
+	checkReadyCtx, cancel := context.WithTimeout(ctx, contextTimeout)
+	defer cancel()
+	checkDaemonsetFunc := func() error {
+		daemonset, err := daemonsets.Get(ctx, ds.Name, metav1.GetOptions{})
+		if err != nil {
+			return errors.Wrap(err, "failed to get daemonset")
+		}
+		numReady := daemonset.Status.NumberReady
+		desiredNumReady := daemonset.Status.DesiredNumberScheduled
+		if numReady != desiredNumReady {
+			return errors.Errorf("daemonset %s in namespace %s has %v pods ready but desired %v", ds.Name, ds.Namespace, numReady, desiredNumReady)
+		}
+		return nil
```

Comment:
`WaitForPodDaemonset` has this functionality with some additional flavor within utils.go